main.cpp

```diff
@@ -191,37 +190,41 @@ IntType::~IntType()
     delete intTypeHeap;
 }
 
-IntType& IntType::add(int rhs)
+IntType& IntType::operator+=( const int other) 
 {
-    *intTypeHeap += rhs;
+    *intTypeHeap += other;
     return *this;
 }
 
-IntType& IntType::subtract(int rhs)
+IntType& IntType::operator-=(const int other)
 {
-    *intTypeHeap -= rhs;
+    *intTypeHeap -= other;
     return *this;
 }
 
-
-IntType& IntType::multiply(int rhs)
+IntType& IntType::operator/=( const int other) 
 {
-    *intTypeHeap *= rhs;
-    return *this;
-}
-
-IntType& IntType::divide(int rhs)
-{
-    if (rhs == 0)
+    if (other == 0)
     {
         std::cout << "error divide by zero" << std::endl;
         *intTypeHeap = 0; 
```

Comment:
I should have caught this before.  there is no need to change the value.  
- [ ] don't change `intTypeHeap` if `rhs == 0`